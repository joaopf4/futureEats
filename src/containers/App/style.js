import styled from "styled-components";

export const AppWrapper = styled.div`
    height: 100vh;
    width: 100vw;
    display: flex;
    align-items: center;
`
export const DisplayWrapper = styled.div`
    border: 1px solid black;
    height: 667px;
    width: 375px; 
    margin: 0 auto;
`
export const StatusBar = styled.div`
    height: 20px;
    width: 100%;
    
`
export const PageWrapper = styled.div`
    height: 647px;
    width: 100%; 
`


