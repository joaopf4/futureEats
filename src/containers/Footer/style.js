import styled from "styled-components";


export const FooterWrapper = styled.div`
    height: 49px;
    width: 100%;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    align-items:center;
    justify-items: center;
    box-shadow: 0 -0.9px 0 0 rgba(0, 0, 0, 0.25);
    z-index: 1;

    svg{
        fill: #b8b8b8;
    }

    svg:nth-child(${props=>props.active}){
        fill: #5cb646;
    }
`