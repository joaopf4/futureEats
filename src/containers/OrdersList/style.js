
import styled from 'styled-components'
import Typography from '@material-ui/core/Typography'

export const OrdersListWrapper = styled.div`
width: 100%;
height: ${({orders}) => (orders ? '50' : 'auto')};
align-items: center;
display: flex;
flex-direction: column;
padding: 12px 4px 0 16px;
`

export const IfEmpty = styled(Typography)`
text-align: center;
font-size: 16px;
padding: 12px 0;
`

export const OrderWrapper = styled.div`
width: 100%;
height: ${({orders}) => (orders ? '42' : '200px')};
overflow-y: scroll;
`

export const RestaurantInfo = styled.div`
width: 100%;
height: 86px;
padding-bottom: 8px;
display: flex;
flex-direction: column;
justify-content: space-evenly;
`

export const RestaurantName = styled(Typography)`
font-size: 16px;
font-weight: 700;
color: #5cb646;
`

export const RestaurantAddress = styled(Typography)`
font-size: 16px;
color: #b8b8b8;
`

export const ExpectedTime = styled(Typography)`
font-size: 16px;
color: #b8b8b8;
`
