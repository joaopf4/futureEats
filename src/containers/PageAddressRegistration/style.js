import styled from 'styled-components'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import { FaAngleLeft } from 'react-icons/fa'

export const Form = styled.form`
align-items: center;
display: flex;
justify-content: center;
flex-direction: column;
`
export const LoginWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
`
export const AdressHeader = styled.div `
    height: 44px;
    width: 100%;
    border-bottom: 1px #b8b8b8 solid;
    padding-left: 16px ;
`
export const MyAddress = styled(Typography)`
    margin-top: 17px;
    padding: 5px 20px 5px 20px;
    font-size: 16px;
    font-weight:  bold;
`
export const Inputs = styled(TextField)`
    margin-top: 16px;
    width: 328px;
    height: 56px;
`
export const SaveButton = styled(Button)`
    width: 328px;
    height: 42px;
    border-radius: 2px;
    margin-top: 16px;
`   
export const BackIcon = styled(FaAngleLeft) `
    margin-top: 12px;
    :active{
    background-color: #b8b8b8;
    border-radius: 6px;
}
`



