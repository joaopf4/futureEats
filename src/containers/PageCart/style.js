import styled from 'styled-components'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import RadioGroup from '@material-ui/core/RadioGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Radio from '@material-ui/core/Radio';


export const CartWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    max-height: 100%;
`

export const Title = styled(Typography)`
    width: 100%;
    font-size: 19px;
    padding: 12px 0;
    text-align: center;
    `

export const DeliveryAddressWrapper = styled.div`
    width: 100%;
    height: 76px;
    padding: 16px 16px;
    background-color: #eeeeee;
    justify-content: space-around;
    display: flex;
    flex-direction: column;
    box-shadow: 0 -0.9px 0 0 rgba(0, 0, 0, 0.25);
`

export const AddressTitle = styled(Typography)`
    font-size: 16px;
    color: #b8b8b8;
    width: 328px;
`

export const Address = styled(Typography)`
    font-size: 16px;
    width: 328px;
`

export const PaymentWrapper = styled.div`
    min-height: ${({orders}) => (orders ? 'none' : '355px')};
    max-height: 100%;
    height: auto;
    width: 100%;
    display: flex;
    flex-direction: column;
    padding: 16px 16px 0 16px;
`

export const DeliveryPrice = styled(Typography)`
    width: 100%;
    font-size: 16px;
    text-align: end;
`

export const TotalPrice = styled.div`
    display: flex;
    width: 100%;
    padding: 14px 0;
`

export const SubTotalTitle = styled(Typography)`
    font-size: 18px;
    width: 50%;
    text-align: start;
`

export const SubTotalPrice = styled(Typography)`
    font-size: 18px;
    width: 50%;
    text-align: end;
    font-weight: 700;
    color: #5cb646;
`

export const PaymentMethod = styled.div`
    width: 100%;
`

export const PaymentTitle = styled(Typography)`
    width: 100%;
    padding: 8px 0;
    border-bottom: solid 1px #000000;
`

export const PaymentRadioGroup = styled(RadioGroup)`
    height: auto;
    width: 100%;
    padding-top: 11px;
`

export const RadioPayment = styled(FormControlLabel)`
    width: 100%;
    height: 18px;
    padding: 14px 0;
    font-size: 16px;
    color: black;
`
export const RadioCustom = styled(Radio)`
    && {
        color: #000000
    }
`
export const ConfirmButton = styled(Button)`
    width: 328px;
    height: 42px;
    border-radius: 2px;
    margin: 14px 0;
    background-color: #5cb646;

    &&:hover{
        background-color: #5cb646;
    }
`   