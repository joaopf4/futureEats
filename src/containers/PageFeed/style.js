import styled from "styled-components";
import Typography from "@material-ui/core/Typography";


export const FeedWrapper = styled.div`
    height: 100%;
    width: 100%;
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 44px 72px 42px 1fr 49px;
    justify-items:center;
    align-items:center;
`

export const LogoWrapper = styled.div`
    border-bottom: 1px solid #b8b8b8;
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: center;
`

export const Logo = styled(Typography)`
    margin: auto;
    font-size: 16px;
`

export const NavBar = styled.div`
    height: 42px;
    width: 100%;
    display: flex;
    align-items: center;
    overflow-x: scroll;
    padding-left: 8px;
`

export const OptionText = styled(Typography)`
    padding: 8px;
    font-size: 18px;

    :nth-of-type(${props => props.activeOption}){
       && {
         color: green;  
       } 
    }
`

export const ContentWrapper = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    overflow-y: scroll;
`