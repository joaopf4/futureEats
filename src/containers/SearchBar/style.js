import styled from "styled-components";
import SearchIcon from '@material-ui/icons/Search';
import OutlinedInput from "@material-ui/core/OutlinedInput";


export const SearchBarInput = styled(OutlinedInput)`
height: 56px;
width: 328px;
`
export const SearchIconCustom = styled(SearchIcon)`
color: #b8b8b8;
`