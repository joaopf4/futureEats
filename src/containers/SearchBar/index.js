import React from "react";
import {SearchBarInput, SearchIconCustom,} from "./style";
import { connect } from "react-redux";
import { push } from "connected-react-router";
import { routes } from "../Router";
import {setSearch} from "../../actions/page"

class SearchBar extends React.Component{
    constructor(props){
        super(props)
        this.state={
            searchInput: ""
        }
    }

    handleInputChange = (event) => {
        const { name, value } = event.target
        this.setState({ [name]: value })
    }

    handleOnKeyDown = (event) => {

        if((event.key === "Enter") && (this.state.searchInput!=="")){
            this.props.setSearch(this.state.searchInput)
            this.props.goToPageSearch();
        }
    }

    render(){
        return(
            <div>
                <SearchBarInput
                    name="searchInput"
                    type="text"
                    placeholder="Restaurante"
                    startAdornment={
                        <SearchIconCustom edge="start" />
                    }
                    onChange={this.handleInputChange}
                    onKeyDown={this.handleOnKeyDown}
                />
            </div>
        )
    }
}

const mapDispatchToProps = dispatch => {
    return {
        goToPageSearch: () => dispatch(push(routes.search)),
        setSearch: (searchFromInput) => dispatch(setSearch(searchFromInput))
    }
}

export default connect(null, mapDispatchToProps) (SearchBar);