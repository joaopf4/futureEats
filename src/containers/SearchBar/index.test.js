import React from 'react'
import { SearchBarInput, SearchIconCustom } from '../SearchBar/style'
import renderer from 'react-test-renderer'

test("Testando styled do SearchBarInput", () => {
    const newSnapShot = renderer.create(<SearchBarInput />).toJSON()

    expect(newSnapShot).toMatchSnapshot();
})
test("Testando styled do SearchIconCustom", () => {
    const newSnapShot = renderer.create(<SearchIconCustom />).toJSON()

    expect(newSnapShot).toMatchSnapshot();
})