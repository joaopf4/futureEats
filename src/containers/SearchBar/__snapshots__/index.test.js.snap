// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando styled do SearchBarInput 1`] = `
<div
  className="MuiInputBase-root-14 MuiOutlinedInput-root-1 sc-bdVaJa dmGsiR"
  onClick={[Function]}
>
  <fieldset
    aria-hidden={true}
    className="MuiPrivateNotchedOutline-root-31 MuiOutlinedInput-notchedOutline-8"
    style={
      Object {
        "paddingLeft": 8,
      }
    }
  >
    <legend
      className="MuiPrivateNotchedOutline-legend-32"
      style={
        Object {
          "width": 0.01,
        }
      }
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </legend>
  </fieldset>
  <input
    className="MuiInputBase-input-24 MuiOutlinedInput-input-9"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
</div>
`;

exports[`Testando styled do SearchIconCustom 1`] = `
<svg
  aria-hidden="true"
  className="MuiSvgIcon-root-33 sc-bwzfXH llBMfO"
  focusable="false"
  role="presentation"
  viewBox="0 0 24 24"
>
  <path
    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
  />
</svg>
`;
